src/account/sync/input_selection.rs

```diff
@@ -13,23 +19,44 @@ pub struct Input {
 }
 
 pub fn select_input(target: u64, mut available_utxos: Vec<Input>) -> crate::Result<Vec<Input>> {
-    if target > available_utxos.iter().fold(0, |acc, address| acc + address.balance) {
+    let total_available_balance = available_utxos.iter().fold(0, |acc, address| acc + address.balance);
+    if target > total_available_balance {
         return Err(crate::Error::InsufficientFunds);
     }
 
+    // Not insufficient funds, but still not possible to create this transaction because it would create dust
+    if target != total_available_balance
+        && target as i64 > (total_available_balance as i64 - DUST_ALLOWANCE_VALUE as i64)
```

Comment:
shouldn't the condition here be `total_available_balance - target < DUST_ALLOWANCE_VALUE`? also this doesn't check if the address has a dust allowance output :(

Reply:
also using `i64` here might lead to a panic if the value doesn't fit.

Reply:
A valid iota amount would always fit since the whole supply is 2779530283277761 and i64::MAX 9223372036854775807

Regarding the condition, it checks the same I think, since we check in the previous `target > total_available_balance`, but I can change it so it's clearer

Reply:
oh you're right, i just understood your condition lol